src/components/network-chart/index.js

```diff
@@ -61,80 +60,104 @@ class NetworkChart extends React.PureComponent<Props> {
-    const {
-      maxNetworkRows,
-      timeRange,
-      threadIndex,
-      networkTimingRows,
-      markers,
-      previewSelection,
-      updatePreviewSelection,
-    } = this.props;
+  _getMarkerPosition = markerStart => {
+    const timeRange = this.props.timeRange;
+    const timeRangeTotal = timeRange.end - timeRange.start;
+
+    let markerPosition =
+      (markerStart - timeRange.start) *
+      (100 - CONTAINER_WIDTH) /
+      timeRangeTotal;
+
+    // the bar shall not overlap the first 150px as this is the thread label area.
+    if (markerPosition < 0) {
+      markerPosition = 0;
+    }
+
+    markerPosition = markerPosition + CONTAINER_WIDTH;
+
+    return markerPosition;
+  };
 
-    // The viewport needs to know about the height of what it's drawing, calculate
-    // that here at the top level component.
-    const maxViewportHeight = maxNetworkRows * ROW_HEIGHT;
+  _getMarkerLength = markerDuration => {
+    const timeRange = this.props.timeRange;
+    const timeRangeTotal = timeRange.end - timeRange.start;
+
+    let markerLength =
+      markerDuration * (100 - CONTAINER_WIDTH) / timeRangeTotal;
+
+    if (markerLength < 0.1) {
+      markerLength = 0.1;
+    }
+
+    return markerLength;
+  };
+
+  _getMarkerStyling = (marker: TracingMarker) => {
+    const markerLength = this._getMarkerLength(marker.dur);
+    const markerPosition = this._getMarkerPosition(marker.data.startTime);
+
+    const markerStyling = {
+      width: markerLength + '%',
+      left: markerPosition + '%',
+    };
+    return markerStyling;
+  };
+
+  // Create row with correct details
+  _renderRow = (nodeId: any, index: number) => {
+    const marker = this.props.markers;
+
+    return (
+      <NetworkChartRow
+        marker={marker[index]}
+        index={index}
+        markerStyle={this._getMarkerStyling(marker[index])}
+        threadIndex={this.props.threadIndex}
+      />
+    );
+  };
+
+  render() {
+    const { markers } = this.props;
 
     return (
       <div className="networkChart">
         <MarkerSettings />
         {markers.length === 0 ? (
           <NetworkChartEmptyReasons />
         ) : (
-          <MarkerChartCanvas
-            key={threadIndex}
-            viewportProps={{
-              timeRange,
-              previewSelection,
-              maxViewportHeight,
-              viewportNeedsUpdate,
-              maximumZoom: this.getMaximumZoom(),
-            }}
-            chartProps={{
-              networkTimingRows,
-              markers,
-              updatePreviewSelection,
-              rangeStart: timeRange.start,
-              rangeEnd: timeRange.end,
-              rowHeight: ROW_HEIGHT,
-              threadIndex,
-            }}
+          <VirtualList
+            className="treeViewBody"
+            items={markers}
+            renderItem={this._renderRow}
+            itemHeight={ROW_HEIGHT}
+            columnCount={1}
+            focusable={true}
+            specialItems={[]}
```

Comment:
Those need proper values as they are not optional, what might be something we want?